enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -0,0 +1,878 @@
+api-approvers:
+  - "@tgeer" ## approver for cert-manager component
+creation-date: 2024-01-22
+last-updated: 2024-05-13
+tracking-link:
+  - https://issues.redhat.com/browse/CM-234
+see-also:
+  - NA
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+# Extend cert-manager-operator to manage istio-csr.
+
+## Summary
+
+This enhancement describes the proposal to extend cert-manager-operator to deploy and manage the `istio-csr`
+solution. `OpenShift Service Mesh` facilitates the security of both intra-cluster and external service
+communications through mTLS. And istio-csr acts as an agent to obtain and renew certificates that
+facilitate mTLS from the `cert-manager`.
+
+## Motivation
+
+cert-manager enables obtaining certificates from [various](https://docs.openshift.com/container-platform/4.15/security/cert_manager_operator/index.html#cert-manager-issuer-types_cert-manager-operator-about)
+certificate authorities and also ensures the validity of certificates by renewing them before they expire.
+`OpenShift Service Mesh` secures intra-cluster and external service connections with TLS encryption. `cert-manager`
+can be used to obtain the required certificates through custom resources `issuer` and `clusterissuer`. `istio-csr`
+is an agent that simplifies the process of obtaining the certificates from the `cert-manager` for
+`OpenShift Service Mesh`. `cert-manager-operator` manages `cert-manager` and extending the operator to manage
+`istio-csr` will help the users to use all the solutions mentioned above effectively and easily.
+
+### User Stories
+
+- As an OpenShift user, I want to have an option to dynamically enable `istio-csr`, so that it can be used only
+  when required.
+- As an OpenShift user, I want to have an option to dynamically configure `istio-csr`, so that only the required
+  features can be enabled.
+
+### Goals
+
+- `cert-manager-operator` to be extended to manage `istio-csr` along with currently managed `cert-manager`.
+- New custom resource `istiocsr` to be made available to enable and configure deployment.
+
+### Non-Goals
+
+- `istio-csr` must be enabled and configured before installing `OpenShift Service Mesh`, which is a
+  limitation of `OpenShift Service Mesh`.
+- Downgrading OpenShift cluster from `istio-csr` supported version to unsupported version is not supported.
+
+## Proposal
+
+`istio-csr` will be managed by `cert-manager-operator` which can be enabled and configured by a user as and
+when required, using new custom resource which will be made available for configuring `istio-csr`.
+
+A new controller will be added to `cert-manager-operator` to manage and maintain the `istio-csr` deployment in
+desired state which can be monitored by user through the status sub-resource of the new custom resource.
+
+`istio-csr` controller will make use of static manifest templates for creating below resources to
+successfully deploy `istio-csr` agent.
+* `certificate.cert-manager.io`
```

Comment:
The list has resources created by cert-manager operator as part for cert-manager installation. Are there any differences that needs to be considered for istio-csr?

Reply:
These are specific to istio-csr, overlapping resources could be merged, but thought to be keep different for istio-csr, so that when istio-csr is disabled just these resources will be cleared.